planet/scripts/item_asset_types.py

```diff
@@ -0,0 +1,56 @@
+import requests
+
+ITEM_TYPE_URL = 'https://api.planet.com/data/v1/item-types/'
+ASSET_TYPE_URL = 'https://api.planet.com/data/v1/asset-types/'
+
+_item_types = None
+_asset_types = None
+
+# Default values here are used as a fallback
+# In case the API fails to respond or takes too long.
+_default_item_types = [
+    "PSScene4Band", "PSScene3Band", "REScene", "SkySatScene",
+    "REOrthoTile", "Sentinel2L1C", "PSOrthoTile", "Landsat8L1G"]
+
+_default_asset_types = [
+    "analytic", "analytic_b1", "analytic_b10", "analytic_b11", "analytic_b12",
+    "analytic_b2", "analytic_b3", "analytic_b4", "analytic_b5", "analytic_b6",
+    "analytic_b7", "analytic_b8", "analytic_b8a", "analytic_b9",
+    "analytic_bqa", "analytic_dn", "analytic_dn_xml", "analytic_ms",
+    "analytic_xml", "basic_analytic", "basic_analytic_b1",
+    "basic_analytic_b1_nitf", "basic_analytic_b2", "basic_analytic_b2_nitf",
+    "basic_analytic_b3", "basic_analytic_b3_nitf", "basic_analytic_b4",
+    "basic_analytic_b4_nitf", "basic_analytic_b5", "basic_analytic_b5_nitf",
+    "basic_analytic_dn", "basic_analytic_dn_nitf", "basic_analytic_dn_rpc",
+    "basic_analytic_dn_rpc_nitf", "basic_analytic_dn_xml",
+    "basic_analytic_dn_xml_nitf", "basic_analytic_nitf", "basic_analytic_rpc",
+    "basic_analytic_rpc_nitf", "basic_analytic_sci", "basic_analytic_xml",
+    "basic_analytic_xml_nitf", "basic_panchromatic_dn",
+    "basic_panchromatic_dn_rpc", "basic_udm", "browse", "metadata_aux",
+    "metadata_txt", "udm", "visual", "visual_xml"
+]
+
+def _get_json_or_raise(url, timeout=0.7):
+    resp = requests.get(url, timeout)
```

Comment:
Do we want to retry here ?

Reply:
That's a good question. It's a trade off between accuracy and sluggishness.

My thinking is that if a request fails the first time, it's pretty likely to fail subsequently, and since we have hard-coded lists to fail over to, we're better off giving up on retrieving the server-side list after one try. You could probably convince me otherwise, though.